profiling/benches/interning_strings.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use criterion::*;
+use datadog_profiling::collections::string_table::wordpress_test_data::WORDPRESS_STRINGS;
+
+/// This version is the one we used before having datadog-alloc.
+#[allow(unused)]
+mod old_version {
+    use datadog_profiling::collections::identifiable::{FxIndexSet, Id, StringId};
+    pub struct StringTable {
+        strings: FxIndexSet<Box<str>>,
+    }
+
+    impl StringTable {
+        pub fn new() -> Self {
+            let mut strings = FxIndexSet::<Box<str>>::default();
+            strings.insert("".into());
+            Self { strings }
+        }
+
+        pub fn intern(&mut self, item: &str) -> StringId {
+            // For performance, delay converting the [&str] to a [String] until
+            // after it has been determined to not exist in the set. This avoids
+            // temporary allocations.
+            let index = match self.strings.get_index_of(item) {
+                Some(index) => index,
+                None => {
+                    let (index, _inserted) = self.strings.insert_full(item.into());
+                    debug_assert!(_inserted);
+                    index
+                }
+            };
+            StringId::from_offset(index)
+        }
+
+        #[inline]
+        #[allow(clippy::len_without_is_empty)]
+        pub fn len(&self) -> usize {
+            self.strings.len()
+        }
+    }
+}
+
+// To benchmark a different implementation, import a different one.
```

Comment:
Make this a `cfg` rather than commenting out a line

Reply:
When you are benchmarking you generally make tweaks to things and run it again. It's not just a static A/B thing we're going to run in CI so you want to automate swapping it out.. So it seems weird to have a cfg for this reason.

I think to use the cfg without making code changes, that would mean making a feature for it. Having a feature just for a benchmarking thing seems odd. Plus it breaks the idea of feature unification (features should be additive).

Do these things make sense? Thoughts?